consensus/src/lib.rs

```diff
@@ -50,11 +68,133 @@ pub enum ConsensusCreationError {
     MiningStopped,
     #[error("Mining failed")]
     MiningFailed,
+    #[error("Staking error")]
+    StakingError(#[from] ConsensusPoSError),
+    #[error("Staking failed")]
+    StakingFailed,
+    #[error("Staking stopped")]
+    StakingStopped,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum GenerateBlockInputData {
+    #[codec(index = 0)]
+    PoW,
+    #[codec(index = 1)]
+    PoS(Box<PoSGenerateBlockInputData>),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct PoSGenerateBlockInputData {
```

Comment:
I think it's better to move all PoS related types and functionality somewhere into `pos` submodule to make lib file clean

Reply:
Nice. I've also shuffled a bit of code around into a separate function to make things a bit clearer.